src/module_wrap.cc

```diff
@@ -1364,10 +1377,34 @@ void ModuleWrap::Resolve(const FunctionCallbackInfo<Value>& args) {
         env, "second argument is not a URL string");
   }
 
+  CHECK(args[2]->IsString());
+  Utf8Value conditions_key_utf8(env->isolate(), args[2]);
+  std::string conditions_key(
+      *conditions_key_utf8, conditions_key_utf8.length());
+  auto conditions_it = conditions_cache_.find(conditions_key);
+  if (conditions_it == conditions_cache_.end()) {
+    // split string by '\n' and add to a new set
+    ConditionSet parsed_conditions;
+    size_t end_pos = -1;
+    do {
+      size_t start_pos = end_pos + 1;
+      end_pos = conditions_key.find('\0', start_pos);
+      size_t len =
+        end_pos == std::string::npos ? end_pos : end_pos - start_pos;
+      parsed_conditions.insert(conditions_key.substr(start_pos, len));
+    } while (end_pos != std::string::npos);
+    conditions_it =
+      conditions_cache_
+        .insert(std::make_pair(conditions_key, parsed_conditions))
+        .first;
+  }
```

Comment:
I think this code would actually be simpler and shorter if it were the direct JS array being iterated, like the Object keys iterations in this file?

Or is there a reason you didn't want to pass in a userland JS array here?

Reply:
If the issue is error handling, it should be fine to abort here though still I think?

A test case for that might be something like - 

```js
defaultResolve(specifier, { ...context, conditions: Object.defineProperty(['a'], '0', { get () { throw new Error('blah') } }) })
```

or something like that.

Reply:
The question is: Do we want to cache the set or do we want to reconstruct it each time? My idea here is that most of the time (custom loaders absent), the code path is just "get the existing set from this string" without ever needing an array.

We could still forward the array as an additional argument instead of parsing the key. But that seems like an opportunity for things to get out of sync..? We could create a safe array from the key (`.split('\n')`) on the JS side but that seems wasteful.

I don't think we have to care about getters because the conditions array will likely be serialized and sent through message channels. So in a final design I don't think getters would survive until this layer (or at least they shouldn't be supported).

Reply:
I can update this to accept both the key and the array but I'm not sure if that makes it a whole lot easier since we trade the string split for a bunch of iteration+type assertion+cast-to-native-string boilerplate.

Reply:
I would still prefer to see this change if possible, but not a blocker.

Reply:
I'd be happy to discuss in a follow-up PR but I'd rather keep the C++ side simple and handle edge cases on the JS side where there's clearer stack traces etc..